### PR TITLE
Feature: notification badge-indicator [debugger]

### PR DESCRIPTION
**Pull request by arpitnath**
**Fixes**: #608 

**Loom**: https://www.loom.com/share/d5c9ea5a713a422eb2db63e3948c418e

**app**: https://tooljet-update-pr.herokuapp.com/apps/b945fce6-1272-4468-b5c0-1ec3c875255d

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
@arpitnath looks good, please resolve the merge conflicts.

**Comment by gondar00** [comment]
One suggestion, the notification badge ideally should not cut out of the sidebar

**Comment by arpitnath** [comment]
Should I reduce the size? or change the position?

**Comment by gondar00** [comment]
Also, can you add a check if the count increases 1000 then show something like 999+, to handle this edge case 
ref: 
image

**Comment by gondar00** [comment]
> Should I reduce the size? or change the position?

I think height 20px width 20px should be good, the trick would be to reduce the font-weight property to something like 400

**Comment by arpitnath** [comment]
> Also, can you add a check if the count increases 1000 then show something like 999+, to handle this edge case
> ref:
> image

yes, also if the number of errors increases 100 we can paginate the logs

File: frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx [hunk 1]
```diff
@@ -56,20 +57,41 @@ export const LeftSidebarDebugger = ({ darkMode, components, errors }) => {
...
+
+        if(open ===  false && errorLogs.length !== unReadErrorCount.read) {
+            setUnReadErrorCount((prev) => {
+                let copy = JSON.parse(JSON.stringify(prev))
+                copy.unread = unReadErrors
+                return copy
+            })
+        }
+        
+        if(open === true) {
```
Comment (gondar00) [inline frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx:71]: I don't think a check is required here it can also be written as `if(open)`

**Comment by arpitnath** [comment]
> > Should I reduce the size? or change the position?
> 
> I think reduce the size to height 20px width 20px should be good, the trick would be to reduce the font-weight property to something like 400

cool, will update in this PR

File: frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx [hunk 2]
```diff
@@ -56,20 +57,41 @@ export const LeftSidebarDebugger = ({ darkMode, components, errors }) => {
...
+    React.useEffect(() => {
+        const unReadErrors = errorLogs.length - unReadErrorCount.read
+
+        if(open ===  false && errorLogs.length !== unReadErrorCount.read) {
+            setUnReadErrorCount((prev) => {
+                let copy = JSON.parse(JSON.stringify(prev))
+                copy.unread = unReadErrors
+                return copy
+            })
+        }
```
Comment (gondar00) [inline frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx:69]: `unReadErrors` should be declared inside the if statement where its being used

Comment (arpitnath) [reply frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx:69]: yes, i was using it in the other block too, but later changed and I guess did not notice, thanks

File: frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx (hunk 1 above)
Comment (arpitnath) [reply frontend/src/Editor/LeftSidebar/SidebarDebugger.jsx:71]: Yep, we don't need this, just can put it in the else block and its all good

**Comment by arpitnath** [comment]
Screenshot 2021-08-28 at 4 34 59 PM
Screenshot 2021-08-28 at 4 34 53 PM

**Comment by arpitnath** [comment]
https://www.loom.com/share/e9aca92a8e1c4560aca77e73ab3ef5bc

**Pull request merged**